lib/source-loader/src/server/abstract-syntax-tree/traverse-helpers.js

```diff
@@ -96,31 +96,41 @@ export function findExportsMap(ast) {
   if (!titleProperty) {
     return { addsMap, idsToFrameworks };
   }
-  const title = titleProperty.value.value;
-
-  estraverse.traverse(ast, {
-    fallback: 'iteration',
-    enter: node => {
-      patchNode(node);
-      if (
-        node.type === 'ExportNamedDeclaration' &&
-        node.declaration &&
-        node.declaration.declarations &&
-        node.declaration.declarations.length === 1 &&
-        node.declaration.declarations[0].type === 'VariableDeclarator' &&
-        node.declaration.declarations[0].id &&
-        node.declaration.declarations[0].id.name &&
-        node.declaration.declarations[0].init &&
-        ['CallExpression', 'ArrowFunctionExpression', 'FunctionExpression'].includes(
-          node.declaration.declarations[0].init.type
-        )
-      ) {
-        const storyName = storyNameFromExport(node.declaration.declarations[0].id.name);
-        const toAdd = handleExportedName(title, storyName, node.declaration.declarations[0].init);
-        Object.assign(addsMap, toAdd);
-      }
-    },
-  });
+  const titleValue = titleProperty.value;
+  let title;
+  // if title is string:
+  if (titleValue.type === 'StringLiteral') {
+    title = titleProperty.value.value;
+  }
+  // if a  tagged template string.
+  if (titleValue.type === 'TemplateLiteral' && titleValue.quasis.length > 0) {
+    title = titleValue.quasis[0].value.raw;
```

Comment:
I suppose the support is actually quite limited?

```
import mdxNotes from '../notes/notes.mdx';
import { DocgenButton } from '../../components/DocgenButton';


const stories = () => 'Stories';

export default {
  title: `Addons/Docs/${stories()}`,
  component: DocgenButton,
};
```

wouldn't actually work?

Reply:
works, I tried the following more real-life examples:
```
const docsTitle = title => `Docs/${title}`;
export default {
  title: `Addons/${docsTitle('Stories')}`,
};
```
or
```
import { DocgenButton } from '../../components/DocgenButton';
export default {
  title: `Addons/Docs/${DocgenButton.displayName}`,
};
```
and similar.

For other cases like expressions, dynamic functions etc - I assume there is some library that does art node value evaluation, but I couldn't find one.

I checked in a better example and made the behavior backwards compatible for non-strings

Reply:
What about `paths.macro`? Can it evaluate that correctly?

```js
import * as React from 'react'
import base from 'paths.macro'

export default {
  title: `${base}`
}

export const example = () => <div />
```

Reply:
The evaluation itself works fine. However there are some rules for titles - it can not finish in `/`.

for example we have:
`import base from 'paths.macro'`

that evaluates to `/stories/addon-docs/`

In the case those will work:
```
title: `Docs${base}Story`,
title: `${base}Story`,
```

However currently this will not work directly (ends in `/`):
```title: `Docs${base}` ```